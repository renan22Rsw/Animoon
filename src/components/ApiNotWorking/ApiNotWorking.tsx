import React from "react";

const ApiNotWorking = () => {
  return <div>ApiNotWorking</div>;
};

export default ApiNotWorking;
