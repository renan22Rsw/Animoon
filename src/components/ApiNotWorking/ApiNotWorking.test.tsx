import { render, screen } from "@testing-library/react";
import ApiNotWorking from "./ApiNotWorking";

describe("ApiNotWorking component", () => {
  it("should render the image", () => {
    render(<ApiNotWorking />);
    const image = screen.getByRole("img");
    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute(
      "src",
      expect.stringMatching(/\/_next\/image\?url=%2Fimg.jpg&w=384&q=75/i)
    );
    expect(image).toHaveAttribute("width", "150");
    expect(image).toHaveAttribute("height", "150");
  });

  it("should render the title", () => {
    render(<ApiNotWorking />);
    const title = screen.getByText(
      "Apologies, but it seems the API is currently unavailable. Please try again later."
    );
    expect(title).toBeInTheDocument();
  });

  it("should match a snapshot", () => {
    const { container } = render(<ApiNotWorking />);
    expect(container).toMatchSnapshot();
  });
});
