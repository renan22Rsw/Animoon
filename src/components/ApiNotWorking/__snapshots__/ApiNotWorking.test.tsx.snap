// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiNotWorking component should match a snapshot 1`] = `
<div>
  <div
    class="my-20"
  >
    <h3
      class=" text-center  font-bold md:text-xl"
    >
      Apologies, but it seems the API is currently unavailable. Please try again later.
    </h3>
    <div
      class="flex justify-center my-8"
    >
      <img
        alt="not-working-api-image"
        data-nimg="1"
        decoding="async"
        height="150"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=384&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=384&q=75 2x"
        style="color: transparent;"
        width="150"
      />
    </div>
  </div>
</div>
`;
